python/dgl/graphbolt/graph_storage/csc_sampling_graph.py

```diff
@@ -280,9 +280,11 @@ def sample_neighbors(
         ), "Fanouts should consist of values that are either -1 or \
             greater than or equal to 0."
         if self.metadata and self.metadata.edge_type_to_id:
-            assert len(self.metadata.edge_type_to_id) == fanouts.size(
-                0
-            ), "Fanouts should have the same number of elements as etypes."
+            assert len(self.metadata.edge_type_to_id) in [
+                fanouts.size(0),
+                1,
+            ], "Fanouts should have the same number of elements as etypes or \
+                should have a length of 1."
```

Comment:
to handle case `fanout=-1`?

Reply:
No, the length of `fanout` could be the length of edge types or **1**, **1** was missed before.

Reply:
do we have dedicated testcase for `1`

Reply:
Yes, in sample neighbors

Reply:
as test already in latest code, CI is throwing assertion exception?

Reply:
No, the tested graph doesn't have metadata.

Reply:
then we could add one which has metadata and access the `1` code path?

Reply:
sure , let me add one